index.js

```diff
@@ -102,6 +106,57 @@ function fastifyPostgres (fastify, options, next) {
     }
   }
 
+  if (!fastify.hasRequestDecorator('pg')) {
+    fastify.decorateRequest('pg', null)
+  }
+
+  fastify.addHook('onRoute', routeOptions => {
+    const useTransaction = routeOptions.pg && routeOptions.pg.transact
+
+    if (useTransaction) {
+      const preHandler = async (req, reply) => {
+        const client = await pool.connect()
+
+        if (name) {
+          if (client[name]) {
```

Comment:
you can't apply the same logic that was applied before, it doesn't make much sense here

Reply:
We resolved this in Slack. It makes sense to be defensive and avoid a namespace that matches a property on the `client` object from pg.